### PR TITLE
chore(cx-api): automatically determine new feature flag version

### PR DESCRIPTION
**Pull request by rix0rrr**
With the new feature flags mechanism, you have to put the `v2` version a flag becomes available in when adding a flag.

One problem: the code hasn't been released yet, so you don't know what version that will be. You can guess it's going to be the next release, but what if the review takes a long time, or a new release was juuuust started by the oncall?

----------------------

Solution: we force you to put in the magic version marker `'V2NEXT'`.

We update the sorting mechanism to sort this always after all v2 versions and before any v3 versions, so it appears at the right place in the report.

At release time, when the actual version number is determined and files are updated and committed back to the repository, we replace the magic marker with the actual version number and regenerate the flag report, so that the actual release version number is put in.

These commits will find their way to the `main` branch when the mergeback commit is created at the end of the release pipeline.

This requires inventing a new `cdk-release` protocol: scripts named `on-bump` will be called as part of creating a release, after the version has been bumped but before the new commit has been created.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/cx-api/lib/private/flag-modeling.ts [hunk 1]
```diff
@@ -45,4 +45,34 @@ export type FlagInfo = FlagInfoBase & (
...
   | { readonly type: FlagType.VisibleContext }
-);
\ No newline at end of file
+);
+
+/**
+ * The magic value that will be substituted at version bump time with the actual
+ * new V2 version.
+ *
+ * Do not import this constant in the `features.ts` file, or the substitution
```
Comment (Naumel) [inline packages/@aws-cdk/cx-api/lib/private/flag-modeling.ts:54]: ❓
Can we have a team tag or marker for such nuggets of wisdom?
Something mentioned in our docs that we could search on?

**Review by Naumel — APPROVED** [review]
LGTM

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -967,9 +967,13 @@ Adding a new flag looks as follows:
 
 1. Define a new const under
    [cx-api/lib/features.ts](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/cx-api/lib/features.ts)
-   with the name of the context key that **enables** this new feature (for
+   with the name of the context key that enables this new feature (for
```
[the remark this answers is not in the document]
Comment (rix0rrr) [reply CONTRIBUTING.md:970]: 't were stress

**Comment by Naumel** [comment]
@Mergifyio update

File: packages/@aws-cdk/cx-api/lib/private/flag-modeling.ts (hunk 1 above)
Comment (rix0rrr) [reply packages/@aws-cdk/cx-api/lib/private/flag-modeling.ts:54]: Ah a special "watch this" marker for comments?

Might be a good idea. Feel free to propose something. This particular one is (somewhat) validated by tests as well.

**Pull request merged**